Storage/Storage/CoreData/IterativeStoreMigrator.swift

```diff
@@ -0,0 +1,220 @@
+                    // ordered array of models), check whether the source model is found
+                    // after the final model.
+                    reverse = model.isEqual(sourceModel)
+                } else {
+                    firstFound = true
+                }
+            }
+
+            if firstFound {
+                modelsToMigrate.append(model)
+            }
+
+            if lastFound {
+                break
+            }
+        }
+
+        // Ensure that the source model is at the start of the list.
+        if reverse {
+            modelsToMigrate = modelsToMigrate.reversed()
+        }
+
+        // Migrate through the list
+        for index in 0...(modelsToMigrate.count - 2) {
+            let modelFrom = modelsToMigrate[index]
+            let modelTo = modelsToMigrate[index + 1]
+
+            // Check whether a custom mapping model exists.
+            guard let migrateWithModel = NSMappingModel(from: nil, forSourceModel: modelFrom, destinationModel: modelTo) ??
+                (try? NSMappingModel.inferredMappingModel(forSourceModel: modelFrom, destinationModel: modelTo)) else {
+                    return false
+            }
+
+            // Migrate the model to the next step
+            DDLogWarn("Attempting migration from \(modelNames[index]) to \(modelNames[index + 1])")
+            
+            guard migrateStore(at: sourceStore, storeType: storeType, fromModel: modelFrom, toModel: modelTo, with: migrateWithModel) == true else {
+                return false
+            }
+        }
+
+        return true
+    }
+}
+
+extension CoreDataIterativeMigrator {
+    private static func migrateStore(at url: URL, storeType: String, fromModel: NSManagedObjectModel, toModel: NSManagedObjectModel, with mappingModel: NSMappingModel) -> Bool {
+        // Build a temporary path to write the migrated store.
+        let fileManager = FileManager.default
+        let tempDestinationURL = url.deletingLastPathComponent().appendingPathComponent("migration").appendingPathComponent(url.lastPathComponent)
+        try? fileManager.removeItem(at: tempDestinationURL.deletingLastPathComponent())
+        try? fileManager.createDirectory(at: tempDestinationURL.deletingLastPathComponent(), withIntermediateDirectories: false, attributes: nil)
+
+        // Migrate from the source model to the target model using the mapping,
+        // and store the resulting data at the temporary path.
+        let migrator = NSMigrationManager(sourceModel: fromModel, destinationModel: toModel)
+        do {
+            try migrator.migrateStore(from: url, sourceType: storeType, options: nil, with: mappingModel, toDestinationURL: tempDestinationURL, destinationType: storeType, destinationOptions: nil)
+        } catch {
+            return false
+        }
+
+        // Move the original source store to a backup location.
+        let backupURL = url.deletingLastPathComponent().appendingPathComponent("backup")
```

Comment:
Nit: To make the `migrateStore()` func a little more manageable, perhaps we could refactor this `Move original source store to a backup location.` logic into a separate func?